src/blocks.js

```diff
@@ -3239,23 +3246,44 @@ BlockMorph.prototype.developersMenu = function () {
     return menu;
 };
 
+BlockMorph.prototype.isHidden = function () {
+    return StageMorph.prototype.hiddenPrimitives[this.selector] === true;
+}
+
 BlockMorph.prototype.hidePrimitive = function () {
-    var ide = this.parentThatIsA(IDE_Morph),
-        dict,
-        cat;
+    var ide = this.parentThatIsA(IDE_Morph);
+
     if (!ide) {return; }
     StageMorph.prototype.hiddenPrimitives[this.selector] = true;
+
+    ide.flushBlocksCache(this.paletteForBlock());
+    ide.refreshPalette();
+};
+
+BlockMorph.prototype.showPrimitive = function () {
+    var ide = this.parentThatIsA(IDE_Morph);
+
+    if (!ide) {return; }
+    delete StageMorph.prototype.hiddenPrimitives[this.selector];
+
+    ide.flushBlocksCache(this.paletteForBlock());
+    ide.refreshPalette();
+};
+
+
+BlockMorph.prototype.paletteForBlock = function () {
+    var dict, palette;
     dict = {
         doWarp: 'control',
         reifyScript: 'operators',
         reifyReporter: 'operators',
         reifyPredicate: 'operators',
         doDeclareVariables: 'variables'
     };
-    cat = dict[this.selector] || this.category;
-    if (cat === 'lists') {cat = 'variables'; }
-    ide.flushBlocksCache(cat);
-    ide.refreshPalette();
+
+    palette = dict[this.selector] || this.category;
+    if (palette === 'lists' || palette === 'other') {palette = 'variables'; }
+    return palette;
```

Comment:
I called this the `palette` and not the `category` b/c really that's where the block 'lives'. 

Also not sure it was worth updating, but similar code exists in `Process.prototype.doChangePrimitiveVisibility`. This method could probably be used there. (And all of this speaks to the fact that I think it might be time for a `PaletteMorph` object that collects all this, but eh....)